fs/smb/client/cifs_debug.c

```diff
@@ -1095,6 +1112,79 @@ static const struct proc_ops cifs_security_flags_proc_ops = {
 	.proc_write	= cifs_security_flags_proc_write,
 };
 
+#ifdef CONFIG_TRUENAS
+static int cifs_zfsacl_flags_proc_show(struct seq_file *m, void *v)
+{
+	seq_printf(m, "0x%x\n", global_zfsaclflags);
+	return 0;
+}
+
+static int cifs_zfsacl_flags_proc_open(struct inode *inode, struct file *file)
+{
+	return single_open(file, cifs_zfsacl_flags_proc_show, NULL);
+}
+
+static ssize_t cifs_zfsacl_flags_proc_write(struct file *file,
+					    const char __user *buffer,
+					    size_t count,
+					    loff_t *ppos)
+{
+	int rc;
+	unsigned int flags, idmap_flags;
+	char flags_string[12] = { 0 };
```

Comment:
Considering that `MODFLAG_ALL_IDMAP` is only half a byte, perhaps we could reduce `flags_string`.